macros/src/actix.rs

```diff
@@ -748,7 +748,20 @@ pub fn emit_v2_definition(input: TokenStream) -> TokenStream {
         ),
     };
 
-    let schema_name = extract_rename(&item_ast.attrs).unwrap_or_else(|| name.to_string());
+    let base_name = extract_rename(&item_ast.attrs).unwrap_or_else(|| name.to_string());
+    let type_params: Vec<&Ident> = generics.type_params().map(|p| &p.ident).collect();
+    let schema_name = if type_params.is_empty() {
+        quote! { #base_name }
+    } else {
+        let type_names = quote! {
+            vec![#(#type_params::name()),*]
+                .iter()
+                .filter(|n| n.is_some())
+                .map(|n| n.as_ref().unwrap().clone()).collect::<Vec<String>>()
```

Comment:
could use filter_map here? also maybe into_iter instead of iter avoiding having to write the clone?

Reply:
hmmm... i get a warning for `into_iter`, but can use `filter_map`.

Reply:
✅